rclcpp/include/rclcpp/node.hpp

```diff
@@ -353,6 +353,9 @@ class Node : public std::enable_shared_from_this<Node>
   size_t
   count_graph_users();
 
+  template<typename CallbackT>
+  void on_param_change(CallbackT && callback);
```

Comment:
I would expect an actual callback function to have such a name (`on_param_change`). For this I think `register_param_change_callback` (or similar) would be better.

Also from the name itself it is not clear that repeated invocations will overwrite previous callback functions. That should at least be documented in a docblock. Maybe there should also be a way to unregister a callback again?